lib/SILOptimizer/IPO/DeadFunctionElimination.cpp

```diff
@@ -374,6 +374,7 @@ class DeadFunctionElimination : FunctionLivenessComputation {
         return false;
       });
     }
+    Module->recalculateVTableEntryCache();
```

Comment:
Instead of recalculating the whole cache you could just erase the dead entries in the loop above